### PR TITLE
[Testing] Enabling ported UITests from Xamarin.UITests to Appium - 14

### PR DESCRIPTION
**Pull request by ghost**
### Description of Change

This PR focuses on re-enabling and updating following tests from the Xamarin.UITest framework to be compatible with Appium. The tests, previously commented out, are reviewed, and modified to ensure they are functional with the Appium framework. We are going to enable tests in blocks in different PRs. This is the 14th group of tests enabled.

Additionally, we removed a duplicate UI test, specifically Issue1023, which was redundant with Issue1023_Forms.

##### Description
This PR introduces new helper methods in the `HelperExtensions` class to simplify common UI interactions across different platforms:

1. `TapBackArrow`: Navigates back in the application by simulating a tap on the platform-specific back navigation button.
2. `TapMoreButton`: Taps the "More" button in the app, with platform-specific logic for Android and Windows.

### TestCases

- **Issue9006**
- **Issue1023_Forms**
- **Issue6217**

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs [hunk 1]
```diff
@@ -18,14 +27,15 @@ public Issue9006(TestDevice testDevice) : base(testDevice)
 		[Category(UITestCategories.Compatibility)]
 		public void ClickingOnTabToPopToRootDoesntBreakNavigation()
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:28]: This test is failing on iOS:
image

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs (hunk 1 above)
Comment (anandhan-rajagopal) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:28]: Resolved the CI failure, also ensured and commited the code changes. 
image
image

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs [hunk 2]
```diff
@@ -18,14 +27,16 @@ public Issue9006(TestDevice testDevice) : base(testDevice)
...
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:28]: This test is failing on iOS (crashing):
`The app was expected to be running still, investigate as possible crash`
image

**Review by jsuarezruiz — DISMISSED** [review]

**Comment by anandhan-rajagopal** [comment]
> @anandhan-rajagopal Could you rebase and fix the conflict? Thanks in advance.

Rebase and resolved conflicts

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs (hunk 2 above)
Comment (anandhan-rajagopal) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:28]: @jsuarezruiz  The CI failure highlights a test case issue: Issue9006 - ClickingOnTabToPopToRootDoesntBreakNavigation. I have run this test locally using both VS Code and the command prompt, and it passed successfully. Could you please review it and advise me on what I may have overlooked to resolve the CI issue?
 
image

Comment (jsuarezruiz) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:28]: This is a more detailed stack trace:

```
OpenQA.Selenium.StaleElementReferenceException : Cached elements 'By.xpath: //android.widget.ImageButton[@content-desc='Navigate up']' do not exist in DOM anymore; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception

   at OpenQA.Selenium.WebDriver.UnpackAndThrowOnError(Response errorResponse, String commandToExecute)
   at OpenQA.Selenium.WebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Appium.AppiumElement.Execute(String commandName, Dictionary`2 parameters)
   at OpenQA.Selenium.Appium.AppiumElement._GetAttribute(String attributeName)
   at OpenQA.Selenium.Appium.AppiumElement.<>c__DisplayClass21_0.<GetAttribute>b__0()
   at OpenQA.Selenium.Appium.AppiumElement.CacheValue(String key, Func`1 getter)
   at OpenQA.Selenium.Appium.AppiumElement.GetAttribute(String attributeName)
   at UITest.Appium.AppiumTouchActions.ElementToClickablePoint(AppiumElement element) in /_/src/TestUtils/src/UITest.Appium/Actions/AppiumTouchActions.cs:line 372
   at UITest.Appium.AppiumTouchActions.<TapElement>g__ProcessException|15_0(<>c__DisplayClass15_0&) in /_/src/TestUtils/src/UITest.Appium/Actions/AppiumTouchActions.cs:line 117
   at UITest.Appium.AppiumTouchActions.TapElement(AppiumElement element) in /_/src/TestUtils/src/UITest.Appium/Actions/AppiumTouchActions.cs:line 108
   at UITest.Appium.AppiumTouchActions.Tap(IDictionary`2 parameters) in /_/src/TestUtils/src/UITest.Appium/Actions/AppiumTouchActions.cs:line 84
   at UITest.Appium.AppiumTouchActions.Execute(String commandName, IDictionary`2 parameters) in /_/src/TestUtils/src/UITest.Appium/Actions/AppiumTouchActions.cs:line 52
   at UITest.Appium.AppiumCommandExecutor.Execute(String commandName, IDictionary`2 parameters) in /_/src/TestUtils/src/UITest.Appium/Actions/AppiumCommandExecutor.cs:line 20
   at UITest.Appium.HelperExtensions.Tap(IUIElement element) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 288
   at UITest.Appium.HelperExtensions.Tap(IApp app, IQuery query) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 38
   at UITest.Appium.HelperExtensions.TapBackArrow(IApp app, IQuery query) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 1781
   at UITest.Appium.HelperExtensions.TapBackArrow(IApp app, String customBackButtonIdentifier) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 1771
   at Microsoft.Maui.TestCases.Tests.Issues.Issue9006.ClickingOnTabToPopToRootDoesntBreakNavigation() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:line 34
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

**Comment by rmarinho** [comment]
can we manually rebase please ?

**Comment by anandhan-rajagopal** [comment]
> can we manually rebase please ?

Rebased the branch

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs (hunk 2 above)
Comment (anandhan-rajagopal) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue9006.cs:28]: @jsuarezruiz  I have committed a fix which addressed the test failure in the CI. Could you please review and provide any feedback or concerns you may have?
image

**Comment by anandhan-rajagopal** [comment]
> Can we rebase this one please?
Rebased the branch.

**Review by jfversluis — APPROVED** [review]

**Pull request merged**